tests/test.c

```diff
@@ -1,61 +1,53 @@
 #include "test.h"
 
-#define TEMP_FILE "/tmp/temp_logs.txt"
-
 #define DELIMITER_WIDTH 80
 
-pid_t output_redirect;          // holds process ID of output redirection process
-FILE *temp_fp;                  // file pointer of open temp file
+pthread_t output_handler_tid;   // holds thread ID of output handler thread
 int save_std_out;               // saved standard output file descriptor to return to normal printing after test
 int pipe_fd[2];                 // read and write ends of the pipe created between the parent and child processes
 char *test_output = NULL;       // holds the contents of temp file
 char *rest_of_test_output;      // holds the rest of the test output as we're performing checks
 int check_num = 0;              // increments to report status each time a check is performed
 char *global_test_name = NULL;  // name of test
 
-// child process sigint handler
-static void child_sigint_handler (int signum) {
-    fclose(temp_fp);   // close temp file
-    close(pipe_fd[0]); // close read end of pipe
-    exit(0);
-}
-
-// child process main function that just allows stdout to go to the temp file and waits for SIGINT
-static void child_main_function () {
+/**
+ * This thread prints output to terminal and also copies it to standard output
+ * It constantly reads from the read end of the pipe, and then prints it to stderr
+ * and stores in test_output. It is canceled by the test in the function end_test().
+ * Arguments:
+ *    args: always NULL
+ * Returns: NULL
+ */
+static void *output_handler(void *args) {
+    size_t curr_size = 1024;
+    size_t num_total_bytes_read = 0;
+    char *curr_ptr = test_output = malloc(curr_size);
```

Comment:
bad style. we should never do multiple assignments on same line

Reply:
👌